vc: fix missing attestations due to doppelganger in epoch 1

arnetheduck opened the pull request:
* log which validator triggers doppelganger
* move activity detection closer to where it's performed, record
aggregation as activity (since it's part of the liveness endpoint)
* vc: check for doppelgangers in epoch 0 also (so that activity in epoch
1 can happen)
* avoid some looping when processing activities

arnetheduck committed:
vc: fix false postive doppelganger message on exit

arnetheduck committed:
log validator that triggers doppelganger

cheatfate commented on beacon_chain/validator_client/doppelganger_service.nim:
```diff
@@ -40,6 +40,9 @@ proc processActivities(service: DoppelgangerServiceRef, epoch: Epoch,
           validator.doppelgangerChecked(epoch)
 
           if item.is_live and validator.triggersDoppelganger(epoch):
```
i think `triggersDoppelganger` is ugly name, because its very hard to see `s` inside `trigger(s)` from first view, and it could be considered as `trigger doppelganger` instead of `triggers doppelganger`.

arnetheduck replied:
have a better suggestion?

arnetheduck committed:
doppelganger fixes

* move activity detection closer to where it's performed, record
aggregation as activity (since it's part of the liveness endpoint)
* vc: check for doppelgangers in epoch 0 also (so that activity in epoch
1 can happen)
* avoid some looping when processing activities

arnetheduck committed:
oops
